From 72d7da13da2824812908276751fd3024db2dd0f8 Mon Sep 17 00:00:00 2001
From: Marius Meisenzahl <meisenzahl@users.noreply.github.com>
Date: Thu, 16 Nov 2023 13:57:08 +0100
Subject: [PATCH] Add support for AppStream 1.0 (#275)

* meson: Add define if AppStream version >= 1.0

* FirmwareReleaseView: Add support for AppStream 1.0
---
 meson.build                        | 6 ++++++
 src/Views/FirmwareReleaseView.vala | 4 ++++
 src/meson.build                    | 2 +-
 3 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 140f19fd..35823ae9 100644
--- a/meson.build
+++ b/meson.build
@@ -14,6 +14,8 @@ libdir = join_paths(prefix, get_option('libdir'))
 
 glib_dep = dependency('glib-2.0', version: '>=2.64.0')
 
+appstream_dep = dependency ('appstream', version: '>=0.12.10')
+
 add_project_arguments(
     '-DGETTEXT_PACKAGE="@0@"'.format(gettext_name),
     language:'c'
@@ -28,6 +30,10 @@ if glib_dep.version().version_compare ('>=2.73.0')
     vala_flags += ['--define', 'HAS_GLIB_2_73']
 endif
 
+if appstream_dep.version().version_compare('>=1.0')
+    vala_flags += ['--define', 'HAS_APPSTREAM_1_0']
+endif
+
 add_project_arguments(vala_flags, language: 'vala')
 
 subdir('data')
diff --git a/src/Views/FirmwareReleaseView.vala b/src/Views/FirmwareReleaseView.vala
index 8e03207c..0ae77652 100644
--- a/src/Views/FirmwareReleaseView.vala
+++ b/src/Views/FirmwareReleaseView.vala
@@ -214,7 +214,11 @@ public class About.FirmwareReleaseView : Gtk.Grid {
 
         summary_label.label = release.get_summary ();
         try {
+#if HAS_APPSTREAM_1_0
+            description_label.label = AppStream.markup_convert (release.get_description (), AppStream.MarkupKind.XML);
+#else
             description_label.label = AppStream.markup_convert_simple (release.get_description ());
+#endif
         } catch (Error e) {
             description_label.label = "";
             warning ("Could not convert markup of release: %s", e.message);
diff --git a/src/meson.build b/src/meson.build
index d044d0c1..c47557ee 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -38,7 +38,7 @@ shared_module(
         dependency('gudev-1.0'),
         dependency('udisks2'),
         dependency('libhandy-1'),
-        dependency ('appstream', version: '>=0.12.10'),
+        appstream_dep,
         meson.get_compiler('vala').find_library('posix'),
         switchboard_dep
     ],
-- 
2.43.0

